File: AdamClassLib/Sample Code/APAX/WinCE/CSharp/APAX-PAC-Sample/APAX-5045/FormSafetySetting.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace APAX_5045
{
    public partial class FormSafetySetting : Form
    {
        public delegate void EventHandler_ApplySafetyValueClick(bool[] bVal);
        public event EventHandler_ApplySafetyValueClick ApplySafetyValueClick;
        int m_iChannelTotal;
        bool[] m_bVal;

        public FormSafetySetting(int iChannelTotal, bool[] bVal)
        {
            InitializeComponent();
            m_iChannelTotal = iChannelTotal;
            m_bVal = bVal;
            // Set init information
            for (int i = 0; i < iChannelTotal; i++)
            {
                ListViewItem lvItem = new ListViewItem(i.ToString());
                lvItem.SubItems.Add(bVal[i].ToString());
                gridviewSafety.Items.Add(lvItem);
            }
        }

        private void gridviewSafety_SelectedIndexChanged(object sender, EventArgs e)
        {
            for (int i = 0; i < gridviewSafety.SelectedIndices.Count; i++)
            {
                if (gridviewSafety.Items[gridviewSafety.SelectedIndices[i]].Selected)
                {
                    int iSelectedItem = gridviewSafety.SelectedIndices[i];

                    txtChannel.Text = gridviewSafety.Items[iSelectedItem].Text;
                    if (Convert.ToBoolean(gridviewSafety.Items[iSelectedItem].SubItems[1].Text))
                        rdbtnOn.Checked = true;
                    else
                        rdbtnOff.Checked = true;
                    return;
                }
            }
        }

        private void btnSet_Click(object sender, EventArgs e)
        {
            if (chbxSelecteAll.Checked)
            {
                for (int idx = 0; idx < m_iChannelTotal; idx++)
                {
                    if (rdbtnOn.Checked)
                        gridviewSafety.Items[idx].SubItems[1].Text = "True";
                    else
                        gridviewSafety.Items[idx].SubItems[1].Text = "False";
                }
                return;
            }
            for (int i = 0; i < gridviewSafety.SelectedIndices.Count; i++)
            {
                if (gridviewSafety.Items[gridviewSafety.SelectedIndices[i]].Selected)
                {
                    int iSelectedItem = gridviewSafety.SelectedIndices[i];
                    if (rdbtnOn.Checked)
                        gridviewSafety.Items[iSelectedItem].SubItems[1].Text = "True";
                    else
                        gridviewSafety.Items[iSelectedItem].SubItems[1].Text = "False";
                    return;
                }
            }
        }
        /// <summary>
        /// Apply setting when user configure safety status
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnApply_Click(object sender, EventArgs e)
        {
            string[] o_szVal = new string[m_iChannelTotal];
            bool[] o_bVal = new bool[m_iChannelTotal];
            for (int i = 0; i < m_iChannelTotal; i++)
            {
                if (gridviewSafety.Items[i].SubItems[1].Text == "True")
                    o_bVal[i] = true;
                else
                    o_bVal[i] = false;
            }
            ApplySafetyValueClick(o_bVal);
        }
    }
}

File: AdamClassLib/Sample Code/APAX/WinCE/VB/APAX-PAC-Sample/APAX-5060PE/FormSafetySetting.vb

Class FormSafetySetting
    Public Delegate Sub EventHandler_ApplySafetyValueClick(ByVal bVal() As Boolean)
    Public Event ApplySafetyValueClick As EventHandler_ApplySafetyValueClick
    Private m_iChannelTotal As Integer

    Private m_bVal() As Boolean

    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Public Sub New(ByVal iChannelTotal As Integer, ByVal bVal() As Boolean)
        MyBase.New()
        InitializeComponent()
        m_iChannelTotal = iChannelTotal
        m_bVal = bVal
        ' Set init information
        Dim i As Integer = 0
        Do While (i < iChannelTotal)
            Dim lvItem As ListViewItem = New ListViewItem(i.ToString)
            lvItem.SubItems.Add(bVal(i).ToString)
            gridviewSafety.Items.Add(lvItem)
            i = (i + 1)
        Loop
    End Sub
 
    ''' <summary>
    ''' Apply setting when user configure safety status
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>


    Private Sub btnApply_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnApply.Click

        Dim o_szVal() As String = New String((m_iChannelTotal) - 1) {}
        Dim o_bVal() As Boolean = New Boolean((m_iChannelTotal) - 1) {}
        Dim i As Integer = 0

        Do While (i < m_iChannelTotal)
            If (gridviewSafety.Items(i).SubItems(1).Text = "True") Then
                o_bVal(i) = True
            Else
                o_bVal(i) = False
            End If
            i = (i + 1)
        Loop

        Try

            RaiseEvent ApplySafetyValueClick(o_bVal)

        Catch ex As Exception
            MessageBox.Show("Delegate function fail.", "Error")
        End Try

        Close()

    End Sub
  

    Private Sub btnSet_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSet.Click
        If chbxSelecteAll.Checked Then

            Dim idx As Integer = 0
            Do While (idx < m_iChannelTotal)
                If rdbtnOn.Checked Then
                    gridviewSafety.Items(idx).SubItems(1).Text = "True"
                Else
                    gridviewSafety.Items(idx).SubItems(1).Text = "False"
                End If
                idx = (idx + 1)
            Loop
            Return
        End If

        Dim i As Integer = 0
        Do While (i < gridviewSafety.SelectedIndices.Count)
            If gridviewSafety.Items(gridviewSafety.SelectedIndices(i)).Selected Then
                Dim iSelectedItem As Integer = gridviewSafety.SelectedIndices(i)
                If rdbtnOn.Checked Then
                    gridviewSafety.Items(iSelectedItem).SubItems(1).Text = "True"
                Else
                    gridviewSafety.Items(iSelectedItem).SubItems(1).Text = "False"
                End If
                Return
            End If
            i = (i + 1)
        Loop
    End Sub

    Private Sub gridviewSafety_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles gridviewSafety.SelectedIndexChanged
        Dim i As Integer = 0

        Do While (i < gridviewSafety.SelectedIndices.Count)
            If gridviewSafety.Items(gridviewSafety.SelectedIndices(i)).Selected Then
                Dim iSelectedItem As Integer = gridviewSafety.SelectedIndices(i)
                txtChannel.Text = gridviewSafety.Items(iSelectedItem).Text
                If Convert.ToBoolean(gridviewSafety.Items(iSelectedItem).SubItems(1).Text) Then
                    rdbtnOn.Checked = True
                Else
                    rdbtnOff.Checked = True
                End If
                Return
            End If
            i = (i + 1)
        Loop
    End Sub
End Class